lib/msal-core/src/authority/B2cAuthority.ts

```diff
@@ -3,26 +3,18 @@
  * Licensed under the MIT License.
  */
 
-import { AadAuthority } from "./AadAuthority";
+import { Authority } from "./Authority";
 import { AuthorityType } from "./Authority";
 import { ClientConfigurationErrorMessage } from "../error/ClientConfigurationError";
-import { UrlUtils } from "../utils/UrlUtils";
+import { AADTrustedHostList } from "../utils/Constants";
 
 /**
  * @hidden
  */
-export class B2cAuthority extends AadAuthority {
+export class B2cAuthority extends Authority {
     public static B2C_PREFIX: String = "tfp";
     public constructor(authority: string, validateAuthority: boolean) {
         super(authority, validateAuthority);
-        const urlComponents = UrlUtils.GetUrlComponents(authority);
-
-        const pathSegments = urlComponents.PathSegments;
-        if (pathSegments.length < 3) {
-            throw ClientConfigurationErrorMessage.b2cAuthorityUriInvalidPath;
-        }
-
-        this.CanonicalAuthority = `https://${urlComponents.HostNameAndPort}/${pathSegments[0]}/${pathSegments[1]}/${pathSegments[2]}/`;
```

Comment:
Is this logic elsewhere (e.g. in the base class), or do we not want to run it at all?

Reply:
We do not want to run it at all. It was breaking the b2c sample as the authority url did not contain enough segments. this.CanonicalAuthority is set in the base class, this was only overwriting it

Reply:
Gotya. My understanding was that canonical authority helped normalize the trailing slash on the url, which I think we still want? (if so, and there isn't a unit test to catch this, we should add one)

Reply:
@jasonnutter is correct, we should make sure we using the same mechanisms to normalize the URLs

Reply:
Right, the url is still normalized by the base class. The only thing I removed is the mandate that the url must contain 3 segments (something that is not present for AAD). However, if we want to leave this as is we can change this to only require 2 segments in order to reflect the b2c authority url format: https://<tenant name>.b2clogin.com/<tenant name>.onmicrosoft.com/<policy name>

Reply:
I think we should get rid of it. I see no value add, and this could be an implementation with assumptions that are no longer true (B2C authority formats changed over the years). The base canonical API in UrlUtils should be enough.